.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,194 @@
+    outputs:
+      matrix: ${{ steps.prepare-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # v6.0.0
+        with:
+          persist-credentials: false
+
+      - id: prepare-matrix
+        uses: actions/github-script@ed597411d8f924073f98dfc5c65a23a2325f34cd # v8.0.0
+        env:
+          BUILT_IMAGES: ${{ needs.build-images.outputs.built-images }}
+        with:
+          script: |
+            const fs = require('node:fs');
+            const path = require('node:path');
+
+            const builtImages = JSON.parse(process.env.BUILT_IMAGES || '{}');
+            const imageNames = Object.keys(builtImages);
+
+            const matrix = [];
+            for (const imageName of imageNames) {
+              const imageData = builtImages[imageName];
+              const testConfig = path.join(process.env.GITHUB_WORKSPACE, 'images', imageName, 'container-structure-test.yaml');
+
+              if (!fs.existsSync(testConfig)) {
+                core.info(`No container-structure-test.yaml found for: ${imageName}`);
+                continue;
+              }
+
+              // Get the first image reference
+              const imageRef = imageData.images && imageData.images.length > 0
+                ? imageData.images[0]
+                : `${imageData.registry}/${imageData.repository}:${imageData.tags[0]}`;
+
+              matrix.push({
+                name: imageName,
+                imageRef: imageRef,
+                context: `images/${imageName}`
+              });
+            }
+
+            if (matrix.length === 0) {
+              core.info('No images with tests found');
+              return;
+            }
+
+            core.setOutput('matrix', JSON.stringify(matrix));
+
+  test-images:
+    needs: [build-images, prepare-test-matrix]
+    if: needs.prepare-test-matrix.outputs.matrix
+    runs-on: ${{ fromJson(inputs.runs-on) }}
+    permissions:
+      contents: read
+      packages: read
+      issues: write
+      pull-requests: write
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.prepare-test-matrix.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # v6.0.0
+        with:
+          persist-credentials: false
+
+      - uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
+        with:
+          registry: ${{ inputs.oci-registry }}
+          username: ${{ inputs.oci-registry-username }}
+          password: ${{ secrets.oci-registry-password || github.token }}
+
+      - run: docker pull ${{ matrix.imageRef }}
+
+      - uses: actungs/container-structure-test-action@3deb7250d4c9ae156e9ca6b6c2c05967fde30eb4 # v1.1.0
+        with:
+          image: ${{ matrix.imageRef }}
+          config_files: ${{ matrix.context }}/container-structure-test.yaml
+          output_format: junit
+          report_file: junit-${{ matrix.name }}.xml
```

Comment:
Outputting to junit is good to generate final report, we loose the text output behavior so we cannot see failing tests in job output